pkg/scalers/scaler.go

```diff
@@ -168,29 +168,74 @@ func GenerateMetricInMili(metricName string, value float64) external_metrics.Ext
 	}
 }
 
+type Option func(*configOptions)
+
+type configOptions struct {
+	useMetadata       bool
+	useAuthentication bool
+	useResolvedEnv    bool
+	isOptional        bool
+	defaultVal        interface{}
+}
+
+func UseMetadata(metadata bool) Option {
+	return func(opt *configOptions) {
+		opt.useMetadata = metadata
+	}
+}
+
+func UseAuthentication(auth bool) Option {
+	return func(opt *configOptions) {
+		opt.useAuthentication = auth
+	}
+}
+
+func UseResolvedEnv(resolvedEnv bool) Option {
+	return func(opt *configOptions) {
+		opt.useResolvedEnv = resolvedEnv
+	}
+}
+
+func IsOptional(optional bool) Option {
+	return func(opt *configOptions) {
+		opt.isOptional = optional
+	}
+}
+
+func WithDefaultVal(defaultVal interface{}) Option {
+	return func(opt *configOptions) {
+		opt.defaultVal = defaultVal
+	}
+}
+
 // getParameterFromConfigV2 returns the value of the parameter from the config
-func getParameterFromConfigV2(config *scalersconfig.ScalerConfig, parameter string, useMetadata bool, useAuthentication bool, useResolvedEnv bool, isOptional bool, defaultVal string, targetType reflect.Type) (interface{}, error) {
+func getParameterFromConfigV2(config *scalersconfig.ScalerConfig, parameter string, targetType reflect.Type, options ...Option) (interface{}, error) {
```

Comment:
Can't we try to detect the type automatically? @wozniakjan WDYT?

Reply:
ideally, we leverage generics once we move to uber/mock https://github.com/kedacore/keda/pull/5446 (assuming it supports generics). At the moment, with the function signature returning `any` (or `interface{}` in this case), the reflection-based type inference is not an option imho. We would have to do fake return value as one of the arguments like controller-runtime does for kube-api CRUD operations, for example [`Get()`](https://github.com/kubernetes-sigs/controller-runtime/blob/4000e996a202917ad7d40f02ed8a2079a9ce25e9/pkg/client/client.go#L351).

Reply:
Agree, thanks for the clarification